app/src/main/java/org/mozilla/focus/widget/DownloadListAdapter.java

```diff
@@ -45,22 +45,29 @@
 
     private static final List<String> SPECIFIC_FILE_EXTENSION
             = Arrays.asList("apk","zip","gz","tar","7z","rar","war");
-    private List<DownloadInfo> mDownloadInfo = new ArrayList<>();
+    private List<DownloadInfo> mDownloadInfo;
     private static final int VIEW_TYPE_EMPTY = 0;
     private static final int VIEW_TYPE_NON_EMPTY = 1;
     private static final int ON_OPENING = 2;
     private Context mContext;
     private int mItemCount = 0;
     private boolean isOpening = false;
+    private boolean isLoading = false;
 
     public DownloadListAdapter(Context context){
         mContext = context;
+        mDownloadInfo = new ArrayList<>();
         loadMore();
         isOpening = true;
     }
 
+    public boolean isLoading(){
+        return isLoading;
+    }
+
     public void loadMore(){
-        DownloadInfoManager.getInstance().query(mItemCount,50,this);
+        DownloadInfoManager.getInstance().query(mItemCount,20,this);
```

Comment:
move pageSize into a constant so we can easily change this.

Reply:
In the meanwhile, I think 50 may be a better choice.

Reply:
I afraid that querying  too mach at once may happen TransactionTooLargeException.